hybrid/battery_stateless.py

```diff
@@ -0,0 +1,213 @@
+        
+        if project_life > 1:
+            self._financial_model.value('system_use_lifetime_output', 1)
+        else:
+            self._financial_model.value('system_use_lifetime_output', 0)
+        self._financial_model.value('analysis_period', project_life)
+
+        if len(self.Outputs.P) == self.site.n_timesteps:
+            single_year_gen = self.Outputs.P
+            self._financial_model.value('gen', list(single_year_gen) * project_life)
+
+            self._financial_model.value('system_pre_curtailment_kwac', list(single_year_gen) * project_life)
+            self._financial_model.value('annual_energy_pre_curtailment_ac', sum(single_year_gen))
+            self._financial_model.value('batt_annual_discharge_energy', [sum(i for i in single_year_gen if i > 0)] * project_life)
+            self._financial_model.value('batt_annual_charge_energy', [sum(i for i in single_year_gen if i < 0)] * project_life)
+            self._financial_model.value('batt_annual_charge_from_system', (0,))
+        else:
+            raise RuntimeError
+
+        self._financial_model.execute(0)
+        logger.info("{} simulation executed".format('battery'))
+
+    @property
+    def system_capacity_kwh(self) -> float:
+        """Battery energy capacity [kWh]"""
+        return self._system_capacity_kwh
+
+    @system_capacity_kwh.setter
+    def system_capacity_kwh(self, size_kwh: float):
+        self._financial_model.value("batt_computed_bank_capacity", size_kwh)
+        self.system_capacity_kwh = size_kwh
+
+    @property
+    def system_capacity_kw(self) -> float:
+        """Battery power rating [kW]"""
+        return self._system_capacity_kw
+
+    @system_capacity_kw.setter
+    def system_capacity_kw(self, size_kw: float):
+        self._financial_model.value("system_capacity", size_kw)
+        self.system_capacity_kw = size_kw
+
+    @property
+    def system_nameplate_mw(self) -> float:
+        """System nameplate [MW]"""
+        return self._system_capacity_kw * 1e-3
+    
+    @property
+    def nominal_energy(self) -> float:
+        """Battery energy capacity [kWh]"""
+        return self._system_capacity_kwh
+
+    @property
+    def capacity_factor(self) -> float:
+        """System capacity factor [%]"""
+        return None
+
+    @property
+    def generation_profile(self) -> Sequence:
+        if self.system_capacity_kwh:
+            return self.Outputs.P
+        else:
+            return [0] * self.site.n_timesteps
+
+    @property
+    def annual_energy_kwh(self) -> float:
+        if self.system_capacity_kw > 0:
+            return sum(self.Outputs.P)
+        else:
+            return 0
+        
+    @property
+    def SOC(self) -> float:
+        if len(self.Outputs.SOC):
+            return self.Outputs.SOC[0]
+        else:
+            return self.initial_SOC
+
+    @property
+    def lifecycles(self) -> float:
+        return self.dispatch
```

Comment:
Should this be `return self.Outputs.lifecycles_per_day` ?

Reply:
Yes, thanks!